src/three-components/CachingGLTFLoader.js

```diff
@@ -37,6 +37,20 @@ export class CachingGLTFLoader {
 
     const gltf = await cache.get(url);
 
-    return gltf.scene ? gltf.scene.clone(true) : null;
+    const model = gltf.scene ? gltf.scene.clone(true) : null;
+
+    // Materials aren't cloned when cloning meshes; geometry
+    // and materials are copied by reference. This is necessary
+    // for the same model to be used twice with different
+    // environment maps.
+    if (model) {
+      model.traverse(object => {
+        if (object.material) {
+          object.material = object.material.clone();
```

Comment:
WDYT if clone let us visit each node and decide how to clone it with a callback @mrdoob ?

Could save us a double traversal, but also might be general enough for other use cases